lib/c/exmagick.c

```diff
@@ -572,3 +586,48 @@ ERL_NIF_TERM exmagick_image_dump_blob (ErlNifEnv *env, int argc, const ERL_NIF_T
   { MagickFree(blob_image); }
   return(enif_make_tuple2(env, enif_make_atom(env, "error"), exmagick_make_utf8str(env, errmsg)));
 }
+
+static
+ERL_NIF_TERM exmagick_convert (ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[])
+{
+  ErlNifBinary utf8;
+  char atom[EXM_MAX_ATOM_SIZE];
+  exm_resource_t *resource;
+
+  EXM_INIT;
+  ErlNifResourceType *type = (ErlNifResourceType *) enif_priv_data(env);
+
+  if (0 == enif_get_resource(env, argv[0], type, (void **) &resource))
+  { EXM_FAIL(ehandler, "invalid handle"); }
+  if (0 == enif_get_atom(env, argv[1], atom, EXM_MAX_ATOM_SIZE, ERL_NIF_LATIN1))
+  { EXM_FAIL(ehandler, "argv[1]: bad argument"); }
+
+  if (strcmp("black_threshold_image", atom) == 0)
+  {
+    char str_value[MaxTextExtent];
+
+    if (0 == exmagick_get_utf8str(env, argv[2], &utf8))
+    { EXM_FAIL(ehandler, "argv[2]: bad argument"); }
+
+    exmagick_utf8strcpy(str_value, &utf8, MaxTextExtent);
+
+    if (0 == BlackThresholdImage(resource->image, &str_value))
```

Comment:
warning: incompatible pointer types passing 'char (*)[2053]' to parameter of type 'const char *' [-Wincompatible-$ointer-types]